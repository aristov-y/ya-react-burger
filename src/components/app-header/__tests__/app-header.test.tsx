import { create, act, ReactTestRenderer } from 'react-test-renderer';
import AppHeader from '../app-header';
import React from 'react';

jest.mock('@ya.praktikum/react-developer-burger-ui-components', () => ({
  Logo: () => 'Logo'
}));
jest.mock('../../header-button');
jest.mock('react-router-dom', () => ({
  Link: ({ children }: React.PropsWithChildren<any>) => (<div data-itemid="Link">{children}</div>)
}))

describe('app-header', () => {
  it('should render', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<AppHeader />);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
