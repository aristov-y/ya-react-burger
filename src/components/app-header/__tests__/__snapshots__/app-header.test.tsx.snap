// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-header should render 1`] = `
<header
  className="app-header"
>
  <nav
    className="app-header-nav"
  >
    HeaderButton
    HeaderButton
  </nav>
  <div
    data-itemid="Link"
  >
    Logo
  </div>
  <div
    className="app-header-profile-wrapper"
  >
    HeaderButton
  </div>
</header>
`;
